grid_map_demos/src/normal_filter_comparison_node.cpp

```diff
@@ -74,32 +76,40 @@ void mapAverageFiltering(
 int main(int argc, char ** argv)
 {
   // Initialize node and publisher.
-  ros::init(argc, argv, "normal_filter_comparison_demo");
-  ros::NodeHandle nh("~");
-  ros::Publisher publisher = nh.advertise<grid_map_msgs::GridMap>("grid_map", 1, true);
+  rclcpp::init(argc, argv);
+  rclcpp::Node node("normal_filter_comparison_demo");
```

Comment:
Can we make this a shared pointer? There have been times I've been burned in certain node operations not really working right because it expects a reference count to be incremented. ROS2 is really finnicky about shared pointers of certain objects. This might be completely fine, but its more defensive.

Reply:
Updated the other demo nodes as well.